app/views/advanced/_advanced_search_facets_as_select.html.erb

```diff
@@ -8,7 +8,7 @@
 
 <%# SPLIT: Going through facet field name & split them up into two arrays & sort alphabetically %>
 <% @main_facets = %w[copyright_combined_label_sim topic_combined_label_sim creator_combined_label_sim date_combined_year_label_ssim location_combined_label_sim workType_label_sim resource_type_label_sim file_format_sim language_label_sim non_user_collections_label_ssim institution_label_sim] %>
-<% @secondary_facets = (facet_field_names - @main_facets).sort %>
+<% @secondary_facets = (facet_field_names - @main_facets).each_with_object({}) { |facet, facet_hash| facet_hash[facet] = facet_field_label(facet) }.sort_by { |k,v| v }.to_h %>
```

Comment:
Does `(facet_field_names - @main_facets).sort_by { |field| facet_field_label(field) }` work? That'd be shorter and more readable imo

Reply:
Maaaaaan it does. I way overcomplicated it 🤣